sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/predicates.scala

```diff
@@ -204,6 +204,7 @@ case class In(value: Expression, list: Seq[Expression]) extends Predicate {
 
   override def children: Seq[Expression] = value +: list
   lazy val inSetConvertible = list.forall(_.isInstanceOf[Literal])
+  lazy val isListEmpty = list.isEmpty
```

Comment:
Do we really need this val?

Reply:
I am using it to be consistent with the current implementation (see the line above)

Reply:
Call list.isEmpty is, in comparison, fast and constant time. It doesn't save much of anything to cache it, and the overhead of a lazy val